zerver/lib/actions.py

```diff
@@ -2609,6 +2609,22 @@ def truncate_topic(topic):
     # type: (Text) -> Text
     return truncate_content(topic, MAX_SUBJECT_LENGTH, "...")
 
+def do_update_buddy_list(user_profile, buddy_profile, should_add):
+    # type: (UserProfile, str, bool) -> None
+
+    record = BuddyList.objects.get_or_create(user=user_profile, buddy=buddy_profile)
```

Comment:
@showell  If should_add is False, then we need to delete existing (user,buddy_user) record from the table. As per my understanding, get_or_create either creates a new record or fetches an existing one. So in this case it should fetch the existing record and then we can delete it. So are you sure we shouldn't call get_or_create here?